modules/metastorage/src/test/java/org/apache/ignite/internal/metastorage/MetaStorageManagerTest.java

```diff
@@ -0,0 +1,297 @@
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for the {@link MetaStorageManager}.
+ */
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class MetaStorageManagerTest {
+    /** */
+    @Mock
+    private MessagingService messagingService;
+
+    /** */
+    private final MetastorageMessagesFactory msgFactory = new MetastorageMessagesFactory();
+
+    /** Holder class. */
+    private static class Node {
+        /** */
+        final MetaStorageManager manager;
+
+        /** */
+        final ClusterNode clusterNode;
+
+        /** */
+        Node(ClusterNode clusterNode, MetaStorageManager manager) {
+            this.manager = manager;
+            this.clusterNode = clusterNode;
+        }
+    }
+
+    /**
+     * Creates a "cluster" consisting of nodes of the given names, where every component is mocked, except for the
+     * {@link MetaStorageManager}.
+     */
+    private List<Node> createCluster(String... names) {
+        List<ClusterNode> nodes = Arrays.stream(names)
+            .map(MetaStorageManagerTest::createNode)
+            .collect(Collectors.toUnmodifiableList());
+
+        // Create the Raft component.
+        var raftManager = mock(Loza.class);
+
+        // Create the Network component.
+        var topologyService = mock(TopologyService.class);
+
+        when(topologyService.allMembers()).thenReturn(nodes);
+
+        var clusterService = mock(ClusterService.class);
+
+        when(clusterService.messagingService()).thenReturn(messagingService);
+        when(clusterService.topologyService()).thenReturn(topologyService);
+
+        // Create a mock Meta Storage config with a poll interval of 1 ms.
+        var pollIntervalMillis = mock(ConfigurationValue.class);
+
+        when(pollIntervalMillis.value()).thenReturn(1L);
+
+        var config = mock(MetaStorageConfiguration.class);
+
+        when(config.startupPollIntervalMillis()).thenReturn(pollIntervalMillis);
+
+        return nodes.stream()
+            .map(node -> {
+                VaultManager vaultManager = mock(VaultManager.class);
+
+                when(vaultManager.name()).thenReturn(CompletableFuture.completedFuture(node.name()));
+
+                var metaStorageManager = new MetaStorageManager(
+                    vaultManager, clusterService, raftManager, msgFactory, config
+                );
+
+                return new Node(node, metaStorageManager);
+            })
+            .collect(Collectors.toUnmodifiableList());
+    }
+
+    /**
+     * Creates a {@code ClusterNode} on a random port.
+     */
+    private static ClusterNode createNode(String name) {
+        int port = ThreadLocalRandom.current().nextInt(0, 10000);
```

Comment:
Any chance of collision? What will happen in that case?

Reply:
this is a unit test, no network components actually start, so the port can even be a fixed value